capstone-backend/src/test/java/com/google/sps/data/BookListTest.java

```diff
@@ -52,27 +50,31 @@ public void testContructors() {
     assertNotEquals(bookList1, bookList2);
 
     // Test Diff
-    assertNotEquals(bookList1.getUserID(), bookList2.getUserID());
-    assertNotEquals(bookList1.getBookshelfID(), bookList2.getBookshelfID());
+    assertEquals(false, bookList1.getUserID().equals(bookList2.getId()));
 
-    assertNotEquals(bookList1.getBookshelfID(), bookList2.getBookshelfID());
+    assertEquals(false, bookList1.getBookshelfID().equals(bookList2.getBookshelfID()));
 
-    assertNotNull(bookList1.getId());
-    assertNull(bookList2.getId());
+    assertNotEquals(bookList1.getCollaboratorsIDs(), bookList2.getCollaboratorsIDs());
+
+    assertEquals(false, bookList1.getCollaboratorsIDs().containsAll(bookList2.getCollaboratorsIDs())
+        && bookList2.getCollaboratorsIDs().containsAll(bookList1.getCollaboratorsIDs()));
+
+    assertEquals(false, bookList1.getId().equals(bookList2.getId()));
   }
 
   @Test
   public void testGetters() {
 
-    assertEquals(USER_ID_1, bookList1.getUserID());
-    assertEquals(BOOKSHELF_ID_1, bookList1.getBookshelfID());
+    assertEquals(true, bookList1.getUserID().equals(USER_ID_1));
+    assertEquals(true, bookList1.getBookshelfID().equals(BOOKSHELF_ID_1));
     assertEquals(COLLABORATORS_IDs_1, bookList1.getCollaboratorsIDs());
-    assertEquals(ID_1, bookList1.getId());
+    assertEquals(true, bookList1.getCollaboratorsIDs().containsAll(COLLABORATORS_IDs_1));
+    assertEquals(true, bookList1.getId().equals(ID_1));
 
-    assertEquals(USER_ID_2, bookList2.getUserID());
-    assertEquals(BOOKSHELF_ID_2, bookList2.getBookshelfID());
+    assertEquals(true, bookList2.getUserID().equals(USER_ID_2));
+    assertEquals(true, bookList2.getBookshelfID().equals(BOOKSHELF_ID_2));
     assertEquals(true, bookList2.getCollaboratorsIDs().isEmpty());
-    assertNull(bookList2.getId());
+    assertEquals(true, bookList2.getId().equals(""));
```

Comment:
For your lines with `assertEquals(true..` I would recommend moving the code inside your `equals(....` to replace `true` because it provides much more information when it fails since it will give you the expected and actual values instead of something like `Expected true: was false`

Reply:
Ok